src/cdk/commands/aslVisualizationCDKManager.ts

```diff
@@ -0,0 +1,59 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { getLogger, Logger } from '../../shared/logger'
+import { AslVisualizationCDK } from './aslVisualizationCDK'
+import { ConstructNode } from '../explorer/nodes/constructNode'
+import { renderGraphCommand } from './renderGraph'
+import { StateMachineGraphCache } from '../../../src/stepFunctions/utils'
+import { AbstractAslVisualizationManager } from '../../../src/stepFunctions/commands/visualizeStateMachine/abstractAslVisualizationManager'
+
+export class AslVisualizationCDKManager extends AbstractAslVisualizationManager {
+
+    public constructor(extensionContext: vscode.ExtensionContext) {
+        super(extensionContext)
+    }
+
+    public override async visualizeStateMachine(
+        globalStorage: vscode.Memento,
+        node: ConstructNode
+    ): Promise<vscode.WebviewPanel | undefined> {
+        const logger: Logger = getLogger()
+        //not sure if I can use this cache
+        const cache = new StateMachineGraphCache()
+        // Attempt to retrieve existing visualization if it exists.
+        const existingVisualization = this.getExistingVisualization(node.label)
+        if (existingVisualization) {
+            existingVisualization.showPanel()
+
+            return existingVisualization.getPanel()
+        }
+
+        // Existing visualization does not exist, construct new visualization
+        try {
+            await cache.updateCache(globalStorage)
+
+            const newVisualization = await renderGraphCommand(node)
+            if (newVisualization) {
+                this.handleNewVisualization(node.label, newVisualization)
+                return newVisualization.getPanel()
+            }
+        } catch (err) {
+            this.handleErr(err, logger)
+        }
+
+        return
+    }
+
+    protected handleNewVisualization(key: string, newVisualization: AslVisualizationCDK): void {
+        this.managedVisualizations.set(key, newVisualization)
```

Comment:
I think we need to be able to handle multiple CDK apps having the same node label. This single visualization manager is created when the extension is activated, and is used to manage all visualizations, including if a new CDK app is used. We should switch this to be a map of identifiers for the CDK app (such as the path to the template file) to a map of strings (node label) to visualizations.